src/main/java/ubc/pavlab/rdp/controllers/SearchController.java

```diff
@@ -284,7 +303,34 @@ public ModelAndView requestGeneAccess( @PathVariable UUID anonymousId,
         return modelAndView;
     }
 
+    /**
+     * This endpoint autocomplete ontology terms.
+     * <p>
+     * Results are unique and ordered as per {@link SearchResult#compareTo(SearchResult)} which first groups results by
+     * match type and then by match.
+     */
+    @ResponseBody
+    @GetMapping("/search/ontology-terms/autocomplete")
+    public Object autocompleteTerms( @RequestParam String query, @RequestParam(required = false) Integer ontologyId, Locale locale ) {
+        if ( ontologyId != null ) {
+            Ontology ontology = ontologyService.findById( ontologyId );
+            if ( ontology == null || !ontology.isActive() ) {
+                return ResponseEntity
+                        .status( HttpStatus.NOT_FOUND )
+                        .contentType( MediaType.TEXT_PLAIN )
+                        .body( String.format( "No ontology with ID %d exists or is active.", ontologyId ) );
```

Comment:
This entity will not be displayed properly in the autocomplete output. We need a JSON payload.